src/hdmf/spec/spec.py

```diff
@@ -210,7 +210,7 @@ def is_region(self):
     {'name': 'dims', 'type': (list, tuple), 'doc': 'the dimensions of this dataset', 'default': None},
     {'name': 'required', 'type': bool,
      'doc': 'whether or not this attribute is required. ignored when "value" is specified', 'default': True},
-    {'name': 'parent', 'type': 'BaseStorageSpec', 'doc': 'the parent of this spec', 'default': None},
+    {'name': 'parent', 'type': '~hdmf.spec.spec.BaseStorageSpec', 'doc': 'the parent of this spec', 'default': None},
```

Comment:
```suggestion
    {'name': 'parent', 'type': 'BaseStorageSpec', 'doc': 'the parent of this spec', 'default': None},
```

Reply:
Using the full module name is not going to work here unfortunately

Reply:
```python
from hdmf.utils import docval
from hdmf.spec import GroupSpec

@docval(
    {"name": "arg1", "type": "~hdmf.spec.spec.BaseStorageSpec", "doc": "doc"},
    is_method=False,
)
def test_docval(**kwargs):
    print(kwargs)


if __name__ == "__main__":
    spec = GroupSpec(name="docval_test", data_type_def="DocvalTest", doc="doc")
    test_docval(spec)
```

```
dev ❯ python test_docval.py
Traceback (most recent call last):
  File "/Users/rly/Documents/NWB/scratch/test_docval.py", line 14, in <module>
    test_docval(spec)
  File "/Users/rly/Documents/NWB/hdmf/src/hdmf/utils.py", line 667, in func_call
    pargs = _check_args(args, kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rly/Documents/NWB/hdmf/src/hdmf/utils.py", line 656, in _check_args
    raise ExceptionType(msg)
TypeError: test_docval: incorrect type for 'arg1' (got 'GroupSpec', expected '~hdmf.spec.spec.BaseStorageSpec')
```

Reply:
I'll try to address this in a separate PR.

Reply:
ah thanks, I missed that one